File: vuln-sandbox/node/taint/path-1hop.js
// Taint: Path Traversal — 1-hop (JS)
// Source: req.query.file → Sink: fs.readFile
// Expected: TruePositive (taint/cwe22)
const express = require('express');
const fs = require('fs');
const app = express();

app.get('/download', (req, res) => {
    const file = req.query.file;                // SOURCE: HTTP request param
    fs.readFile(`/uploads/${file}`, (err, data) => { // SINK: file read
        res.send(data);
    });
});

File: vuln-sandbox/python/taint/path-1hop.py
# Taint: Path Traversal — 1-hop (Python)
# Source: request.args.get('file') → Sink: open()
# Expected: TruePositive (taint/cwe22)
from flask import Flask, request

app = Flask(__name__)

@app.route('/download')
def download():
    filename = request.args.get('file')        # SOURCE: HTTP request param
    with open(f"/uploads/{filename}") as f:    # SINK: file open
        return f.read()
